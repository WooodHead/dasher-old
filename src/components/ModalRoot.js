import React from 'react';
import { connect } from 'react-redux';
import DeleteBoardModal, { DELETE_BOARD_MODAL } from './DeleteBoardModal';

const MODAL_COMPONENTS = {
  [DELETE_BOARD_MODAL]: DeleteBoardModal,
};

const ModalRoot = ({ modalType, modalProps }) => {
  if (!modalType) {
    return null;
  }

  return React.createElement(MODAL_COMPONENTS[modalType], modalProps);
};

export default connect(state => state.modal)(ModalRoot);
