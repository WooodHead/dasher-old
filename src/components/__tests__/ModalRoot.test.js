import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'unstated'

import ModalContainer from '../../containers/ModalContainer'
import ModalRoot from '../ModalRoot'

test('renders nothing when modal component is unset', () => {
  const modal = new ModalContainer()

  const tree = renderer
    .create(
      <Provider inject={[modal]}>
        <ModalRoot />
      </Provider>,
    )
    .toJSON()

  expect(modal.state.component).toBeNull()
  expect(modal.state.props).toEqual({})
  expect(tree).toMatchSnapshot()
})

test('renders modal component after showModal() is called', () => {
  const modal = new ModalContainer()
  const ExampleComponent = props => <div {...props} />
  const exampleProps = { foo: 'bar', baz: 42, children: 'Example' }

  modal.showModal(ExampleComponent, exampleProps)

  const tree = renderer
    .create(
      <Provider inject={[modal]}>
        <ModalRoot />
      </Provider>,
    )
    .toJSON()

  expect(modal.state.component).toBe(ExampleComponent)
  expect(modal.state.props).toBe(exampleProps)
  expect(tree).toMatchSnapshot()
})
