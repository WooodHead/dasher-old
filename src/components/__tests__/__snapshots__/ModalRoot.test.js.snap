// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal component after showModal() is called 1`] = `
<div
  baz={42}
  foo="bar"
  hideModal={[Function]}
>
  Example
</div>
`;

exports[`renders nothing when modal component is unset 1`] = `null`;
